en/computer_vision/path_planning_interface.md

```diff
@@ -0,0 +1,90 @@
+- `time_usec`: UNIX Epoch time.
+- `valid_points`: 3
+- Point 0 - Current waypoint *type adapted* by FlightTaskAutoMapper (see [notes below](#type_adapted)):
+  - `pos_x[0]`, `pos_y[0]`, `pos_z[0]`: Type adapted x-y-z NED local position of *current* mission waypoint.
+  - `vel_x[0]`, `vel_y[0]`, `vel_z[0]`: Type adapted x-y-z NED local velocity of *current* mission waypoint.
+  - `acc_x[0]`, `acc_y[0]`, `acc_z[0]`: NaN
+  - `pos_yaw[0]`: Current yaw angle
+  - `vel_yaw[0]`: NaN
+  - `command[0]`: The [MAVLink Command](https://mavlink.io/en/messages/common.html#mavlink-commands-mavcmd) for the current waypoint. 
+- Point 1 - Current waypoint (Unmodified/not type adapted)):
+  - `pos_x[1]`, `pos_y[1]`, `pos_z[1]`: x-y-z NED local position of *current* mission waypoint
+  - `vel_x[1]`, `vel_y[1]`, `vel_z[1]`: NaN
+  - `acc_x[1]`, `acc_y[1]`, `acc_z[1]`: NaN
+  - `pos_yaw[1]`: Yaw setpoint
+  - `vel_yaw[1]`: Yaw speed setpoint
+  - `command[1]`: The [MAVLink Command](https://mavlink.io/en/messages/common.html#mavlink-commands-mavcmd) for the current waypoint.
+- Point 2 - Next waypoint in local coordinates (unmodified/not type adapted):
+  - `pos_x[2]`, `pos_y[2]`, `pos_z[2]`: x-y-z NED local position of *next* mission waypoint
+  - `vel_x[2]`, `vel_y[2]`, `vel_z[2]`: NaN
+  - `acc_x[2]`, `acc_y[2]`, `acc_z[2]`: NaN
+  - `pos_yaw[2]`: Yaw setpoint
+  - `vel_yaw[2]`: Yaw speed setpoint
+  - `command[2]`: The [MAVLink Command](https://mavlink.io/en/messages/common.html#mavlink-commands-mavcmd) for the next waypoint.
+- All other indices/fields are set as NaN.
+
+<span id="type_adapted"></span>
+Notes:
+- Point 0 is the current waypoint/target modified based on the type of target.
+  For example, it makes sense when landing to specify the target x, y coordinates and a descent velocity.
+  To achieve this `FlightTaskAutoMapper` modifies land waypoints in Point 0 to set the z component of position to NAN and the z-velocity to a desired value.
+- Point 1 and 2 are not used by the safe landing planner.
+- Point 1 is used by local and global planner.
+
+On the companion side, MAVROS translates the MAVLink message into ROS messages, which are eventually handled by the appropriate planner.
+The planner plans a path to the waypoint/target, and sends it to the vehicle as a stream of `TRAJECTORY_REPRESENTATION_WAYPOINTS` messages that have the setpoint in Point 0.
+
+The fields for the `TRAJECTORY_REPRESENTATION_WAYPOINTS` message from the companion computer are set as shown:
+- `time_usec`: UNIX Epoch time.
+- `valid_points`: 1
+- Current vehicle information:
+  - `pos_x[0]`, `pos_y[0]`, `pos_z[0]`: x-y-z NED vehicle local position setpoint
+  - `vel_x[0]`, `vel_y[0]`, `vel_z[0]`: x-y-z NED velocity setpoint
+  - `acc_x[0]`, `acc_y[0]`, `acc_z[0]`: NaN
+  - `pos_yaw[0]`: Yaw angle setpoint
+  - `vel_yaw[0]`: Yaw speed setpoint
+  - `command[0]`: NaN.
+- All other indices/fields are set as NaN.
+
+PX4 will stream trajectory messages ... TBD.
```

Comment:
@mrivi The main remaining TBD is how the systems handle fail cases.
1. If I enable COM_OBS_AVOID and go into mission mode, what happens if I get no heartbeat from companion? What if I have heartbeat and setpoints messages, but then they drop out?
   - I assume that in either case there will be some wait period, and then a failsafe. I further assume that the failsafe for land would be to go to normal Land mode, and failsafe for obstacle avoidance would be to drop to normal missions. Is this the case? If so, is this immediate or is there a wait period?
   - What about if you're in land mode on PX4 but the companion is set up for obstacle avoidance. You'll have a live HEARTBEAT and be sending messages to the companion. The companion will know that it isn't able to support land mode and will not send messages back. What will PX4 do?

Reply:
If the planner stops sending HEARTBEAT messages while in an AUTO mode the vehicle will switch to LOITER. It will also switch to LOITER if the trajectory messages stop being sent.

I don't believe there is much more in way of edge case handling.

Reply:
If the HEARTBEAT stops coming in, a message will show up in QGC say "Avoidance system lost" or "Avoidance system timeout" dipending on the MAV_STATE. However no failsafe action is taken.

If the trajectory message is older than 0.5 seconds and the vehicle is in auto with COM_OBS_AVOID=1, the vehicle will switch into Loiter.